src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs

```diff
@@ -605,17 +602,20 @@ public async ValueTask VerifyFileHash(CancellationToken token) {
 		var stream = reader.Stream;
 		var footer = _chunkFooter;
 
-		byte[] hash;
-		using (var md5 = MD5.Create()) {
-			// hash whole chunk except MD5 hash sum which should always be last
-			await MD5Hash.ContinuousHashFor(md5, stream, 0, _fileSize - ChunkFooter.ChecksumSize, token);
-			md5.TransformFinalBlock([], 0, 0);
-			hash = md5.Hash;
-		}
+		using var md5 = IncrementalHash.CreateHash(HashAlgorithmName.MD5);
+
+		// hash whole chunk except MD5 hash sum which should always be last
+		await MD5Hash.ContinuousHashFor(md5, stream, 0, _fileSize - ChunkFooter.ChecksumSize, token);
+		VerifyHash(footer.MD5Hash, md5);
 
-		// Perf: use hardware accelerated byte array comparison
-		if (!footer.MD5Hash.SequenceEqual(hash))
-			throw new HashValidationException();
+		static void VerifyHash(ReadOnlySpan<byte> expected, IncrementalHash actual) {
+			Span<byte> buffer = stackalloc byte[ChunkFooter.ChecksumSize];
+
+			// Perf: use hardware accelerated byte array comparison
+			if (!actual.TryGetCurrentHash(buffer, out var bytesWritten)
```

Comment:
is it better to use `actual.GetCurrentHash(...)` instead of `actual.TryGetCurrentHash(...)` here ? as we guarantee that the destination buffer is large enough to hold the checksum.

Reply:
From perf point of view, they are the same. `GetCurrentHash` throws an exception if the size is not enough, we can avoid that branch.

Reply:
Fixed